source/MaterialXShaderGen/ShaderGenerators/Glsl/GlslShaderGenerator.cpp

```diff
@@ -135,6 +139,27 @@ GlslShaderGenerator::GlslShaderGenerator()
     // <!-- <texcoord> -->
     registerImplementation("IM_texcoord__vector2__glsl", TexCoordGlsl::creator);
     registerImplementation("IM_texcoord__vector3__glsl", TexCoordGlsl::creator);
+    // <!-- <geomcolor> -->
+    registerImplementation("IM_geomcolor__float__glsl", GeomColorGlsl::creator);
+    registerImplementation("IM_geomcolor__color2__glsl", GeomColorGlsl::creator);
+    registerImplementation("IM_geomcolor__color3__glsl", GeomColorGlsl::creator);
+    registerImplementation("IM_geomcolor__color4__glsl", GeomColorGlsl::creator);
+    // <!-- <geomattrvalue> -->
+    registerImplementation("IM_geomattrvalue__integer__glsl", GeomAttrValueGlsl::creator);
+    registerImplementation("IM_geomattrvalue__boolean__glsl", GeomAttrValueGlsl::creator);
```

Comment:
Guess this is where I'm still not clear. I don't think we'd ever have a stream of bools or strings so would we still end up interpreted this as per vertex data?

Reply:
Yep, changing to uniform instead.

This raises another question though, if there is a need for an additional node that can handle named vertex streams? Maybe a new <primvar> node would be needed. We can check with ILM.

Reply:
The node is called <geomattrvalue> in the specification, so I think it's better to use the same naming for the implementation here to be consistent. And according to spec its used for constant values associated with geometry, so using uniforms for this in GLSL seems appropriate.

Having support for custom vertex streams becomes a separate thing then, and for that we would need a new node added to the specification (like the <primvar> node I suggested above). But needs to be discussed with Lucasfilm, and we should add a separate task for that..